New package: weather-2.0

dicktyr opened the pull request.

Vaelatern commented on srcpkgs/weather/template:
```diff
@@ -0,0 +1,38 @@
+pkgname=weather
+version=2.0
+revision=1
+# upstream supports both python2 and python3
```
It is expected you have a different subpackage for different versions of python. https://github.com/voidlinux/void-packages/blob/master/Manual.md#pkgs_python

dicktyr replied:
it's not clear to me how to separate the two (2.7 from 3.4) using only a subpackage

i.e. if the 3.4 subpackage is installed
the 2.7 will also be installed because it's in the main package
but maybe somebody wants only the 3.4

Vaelatern replied:
There are examples in the srcpkgs

dicktyr replied:
yes I've been looking for one

Vaelatern replied:
Good example is `python-urllib`

dicktyr replied:
thanks
but that doesn't answer my question

it's not a python-module so I assume I need do_install
and that's why it's not clear to me
how the two versions should be separated

stpx replied:
@dicktyr I think you can just choose the python version of your preference. If or when someone wants/needs the other version, we can add a subpackage.

dicktyr replied:
ok thanks
I don't mind making it work for both
but I need more guidance to do that

`./xbps-src pkg weather` fails with

    => ERROR: python-2.7.11_1: failed to install 'libressl-devel-2.2.5_1' dependency! (error 19)

stpx replied:
Your repo is outdated: 
```sh
$ git pull --rebase upstream master
```

dicktyr replied:
thanks
I'm learning about git and github as well

Vaelatern commented on srcpkgs/weather/template:
```diff
@@ -0,0 +1,38 @@
+revision=1
+# upstream supports both python2 and python3
+noarch=yes
+depends="python"
```
"To allow cross compilation, the python-devel package (for python 2.7) must be added to hostmakedepends and makedepends. If any other python version is also supported, for example python3.4, those must also be added as host and target build dependencies."

Vaelatern commented on srcpkgs/weather/template:
```diff
@@ -0,0 +1,38 @@
+# upstream supports both python2 and python3
+noarch=yes
+depends="python"
+pycompile_module="weather.py"
```
It's not byte compiled. @xtraeme doesn't this go against documentation to NOT have this byte-compiled? Or am I reading the manual wrong?

stpx replied:
This line is correct.

Python code is byte-compiled, during the package configure phase, after the package has been unpacked via `xbps-install(1)`; or manually via `xbps-reconfigure(1)`.

Vaelatern replied:
I saw, when I tested an install, a `weather.py` file that doesn't look like a `.pyc` file or byte compiled. Perhaps I saw it wrong.

Vaelatern commented on srcpkgs/weather/template:
```diff
@@ -0,0 +1,38 @@
+do_install() {
+	vbin weather
+	vinstall weather.py 644 usr/lib/python2.7/site-packages
+	vinstall weather.py 644 usr/lib/python3.4/site-packages
```
Again, different subpackages for different python versions.

dicktyr commented:
I believe I've addressed all comments/concerns
but this request remains unaccepted

it's also coming from my master branch
(because I had not yet learned to isolate
each pull request in a separate branch)
and I've been holding back other pull requests
until it is accepted or rejected
